### DISCUSSION TITLE
Feature Request: Allow Manual Configuration of Requests Per Minute

### DISCUSSION THREAD

**Discussion by cal88** [cline/cline#588] [Feature Requests]
I would like to request the addition of a feature that enables users to manually set the number of requests per minute within the cline's settings. This feature will enhance the current usability for a broader range of models as currently available, including those using limited or free API plans. By providing manual control over request rates, cline can better cater to diverse user needs and prevent common issues related to rate limiting.

**Use Cases**

1. Lower Tier APIs: Users with subscriptions to lower tier APIs often face stricter rate limits. Manually setting the requests per minute would help them stay within their allowed usage and avoid potential throttling.
2. Free LLM Models: Users leveraging free language model services typically have lower requests per minute allowances. This feature would assist in managing their usage effectively, ensuring uninterrupted access.

**Benefits**

- Flexibility: Accommodates a wider range of API plans, from free tiers to premium subscriptions.
- User Control: Empowers users to tailor the extension's behavior according to their specific API rate limits.
- Prevents Overuse: Helps avoid exceeding rate limits, which can lead to service interruptions or additional costs.

**Implementation Suggestions**

- Settings Integration: Add a configurable option in the extension's settings panel where users can input their desired requests per minute.
- Default Values: Provide sensible default values based on common API rate limits, while allowing users to override them as needed.

**Comment by kslamph** [comment]
this is definitely a critical feature, a deal breaker for use of many models , providers. without this feature, the real usable models are very limited.

**Reply by tariqs0** [reply]
was this implemented or not yet?

**Reply by assaf-b** [reply]
> was this implemented or not yet?

Unfortunately not yet.

**Comment by assaf-b** [comment]
Critical feature, please try and add it

**Comment by stevejford** [comment]
With Gemini Rate and OpenRouter rate limiting the new models this would be
a life saver

**Comment by Sophanos** [comment]
Any update on this? Really needed

**Comment by Slowly-Grokking** [comment]
This is a must-have feature.  It's the only thing that has be thinking of retrying Roo

**Comment by Mobinake** [comment]
im really waiting to this be a feature.

**Comment by piyushgoel6124** [comment]
i am a student and have been using cline since last 3 months and really need this feature or maybe some hack to do this in the internal settings of cline as gemini models apply rate limiting very hard

**Comment by epurn** [comment]
Also waiting on this, not having it renders GPT5 unusable
